driver/sum_driver.hpp

```diff
@@ -0,0 +1,334 @@
+    std::vector<Tgpu> out;
+    std::vector<Tref> outhost;
+
+    size_t ws_sizeInBytes;
+
+    int dim;
+    miopenSumNanPropagation_t nanPropagation;
+};
+
+template <typename Tgpu, typename Tref>
+int SumDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int SumDriver<Tgpu, Tref>::GetandSetData()
+{
+    std::vector<int> in_len = GetInputTensorLengthsFromCmdLine();
+    dim                     = inflags.GetValueInt("DimToReduce");
+
+    SetTensorNd(inputDesc, in_len, data_type);
+
+    std::vector<int> out_len;
+
+    for(int i = 0; i < in_len.size(); i++)
+    {
+        if(i != dim)
+        {
+            out_len.push_back(in_len[i]);
+        }
+    }
+
+    SetTensorNd(outputDesc, out_len, data_type);
+
+    nanPropagation = static_cast<miopenSumNanPropagation_t>(inflags.GetValueInt("NanPropagation"));
+
+    return 0;
+}
+
+template <typename Tgpu, typename Tref>
+int SumDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only Forward Sum (Default=1)", "int");
+    inflags.AddInputFlag("InputDimLengths",
+                         'D',
+                         "256,4,8732",
+                         "The dimensional lengths of the input tensor(Default=256,4,8732)",
+                         "string");
+    inflags.AddInputFlag(
+        "DimToReduce", 'R', "1", "The indice of the dimensions to be reduced(Default=1)", "int");
+    inflags.AddInputFlag("NanPropagation",
+                         'N',
+                         "0",
+                         "Nan number propagation mode (check the miopenSumNanPropagation_t in "
+                         "miopen.h) (Default=0 to indicate no Nan propagation)",
+                         "int");
+    inflags.AddInputFlag("iter", 'i', "10", "Number of Iterations (Default=10)", "int");
+    inflags.AddInputFlag("verify", 'V', "1", "Verify Each Layer (Default=1)", "int");
+    inflags.AddInputFlag("time", 't', "0", "Time Each Layer (Default=0)", "int");
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time Each Layer, Requires time == 1 (Default=0)", "int");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+std::vector<int> SumDriver<Tgpu, Tref>::GetInputTensorLengthsFromCmdLine()
+{
+    std::string lengthsStr = inflags.GetValueStr("InputDimLengths");
+
+    std::vector<int> lengths;
+    std::size_t pos = 0;
+    std::size_t new_pos;
+
+    new_pos = lengthsStr.find(',', pos);
+    while(new_pos != std::string::npos)
+    {
+        std::string sliceStr = lengthsStr.substr(pos, new_pos - pos);
+
+        int len = std::stoi(sliceStr);
+
+        lengths.push_back(len);
+
+        pos     = new_pos + 1;
+        new_pos = lengthsStr.find(',', pos);
+    };
```

Comment:
[Q] Could lengthsStr be in the wrong format?

Reply:
fixed it